### PR TITLE
recover-eu

### PR DESCRIPTION
**Pull request by ekamau**
draft parser for recover EU

### CODE REVIEW

File: isaric/parsers/recover-eu_parser.toml [hunk 1]
```diff
@@ -0,0 +1,325 @@
+#:schema ../../schemas/dev/parser.schema.json
+
+[adtl]
+  name = "recover eu"
+  description = "recover EU consortium"
+
```
Comment (pipliggins) [inline isaric/parsers/recover-eu_parser.toml:6]: defaultDateFormat = "%m/%d/%Y"

File: isaric/parsers/recover-eu_parser.toml [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    aggregation = "lastNotNull"
+    schema = "../../schemas/dev/subject.schema.json"
+
+  [adtl.tables.visit]
+    kind = "groupBy"
+    groupBy = "visit_id"
+    aggregation = "lastNotNull"
+    schema = "../../schemas/dev/visit.schema.json"
+    optional-fields = [ "outcome", "country_iso3" ]
+
```
Comment (pipliggins) [inline isaric/parsers/recover-eu_parser.toml:19]: Add date_outcome - both date of death and discharge date are empty in the data (unless Sadie finds a field that isn't empty for this)

File: isaric/parsers/recover-eu_parser.toml [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    ]
+    
+  [subject.earliest_admission_date]
+    combinedType = "min"
+    fields = [{ field = "adm_dt", description = "Date of hospital admission:" }]
+
+  [subject.age]
+    field = "DOB"
+    description = "Date of birth:"
+    apply = { function = "yearsElapsed", params = ["$adm_dt"] }
```
Comment (pipliggins) [inline isaric/parsers/recover-eu_parser.toml:54]: apply = { function = "yearsElapsed", params = [
      "$adm_dt",
      2022,
      "%m/%d/%Y",
      "%m/%d/%Y",
    ] }

File: isaric/parsers/recover-eu_parser.toml [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+[subject.has_asplenia]
+  field = "CIMMUN_ASPL"
+  ref = "Y/N"
+
+[subject.has_comorbidity_other] # type - array
+  combinedType = "set"
+    excludeWhen = "none"
+    fields = [
+      { field = "chr_other" },
+      { field = "COTH_OTHER" },
```
Comment (pipliggins) [inline isaric/parsers/recover-eu_parser.toml:176]: These two have numerical (1/2 and 0/1) contents - maybe they're housekeeping variables for the later options? regardless, they shouldn't be in here. COTH_OTHER particularly is just a yes/no flag for COTH_OTHER_SPEC I think

File: isaric/parsers/recover-eu_parser.toml
Comment (sadiekelly) [inline isaric/parsers/recover-eu_parser.toml:1]: @ekamau Here's my review of recover-eu, thanks!
[RECOVER EU parser review 2023-09-06.xlsx](https://github.com/globaldothealth/isaric/files/12536789/RECOVER.EU.parser.review.2023-09-06.xlsx)

Comment (pipliggins) [reply isaric/parsers/recover-eu_parser.toml:1]: Thanks @sadiekelly - @ekamau I've made the changes Sadie suggested, you just need to go through and check you agree, then we can merge this.

**Review by ekamau — COMMENTED** [review]
I have looked at Sadie's comments and the edits look good @pipliggins - thanks very much!

**Pull request merged**